app/models/miq_request_workflow.rb

```diff
@@ -1042,7 +1042,10 @@ def ems_folder_to_hash_struct(ci)
   end
 
   def storage_to_hash_struct(ci)
-    build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type])
+    storage_cluster = ci.storage_clusters.detect{ |cluster| cluster.ems_id == get_source_and_targets[:ems].try(:id) }
```

Comment:
Just tried adding a datastore that is shared between datacenters and get this error from vsphere, `Storage DRS does not allow shared datastores across datacenters.` so we are going to be good just a check on the ems_id.  A datastore will only ever be able to be in one StorageCluster per VC.